src/types.cpp

```diff
@@ -332,24 +331,24 @@ timestamp_t RTimestampType::Convert(double val) {
 	return Timestamp::FromEpochMicroSeconds(round(val * Interval::MICROS_PER_SEC));
 }
 
-dtime_t RTimeSecondsType::Convert(double val) {
-	return dtime_t(int64_t(val * Interval::MICROS_PER_SEC));
+interval_t RIntervalSecondsType::Convert(double val) {
+	return Interval::FromMicro(int64_t(val * Interval::MICROS_PER_SEC));
```

Comment:
This is consistent with the old implementation, but when converting datetimes I think we're rounding. Should we change this as part of the change here and further below?

```suggestion
	return Interval::FromMicro(int64_t(round(val * Interval::MICROS_PER_SEC)));
```